templates/raspbian/resources/usr/share/initramfs-tools/scripts/cattlepi-base/helpers

```diff
@@ -124,6 +124,61 @@ cattlepi_fetch_update_config()
     cattlepi_fetch_update_config_invoked=yes
 }
 
+# apply the requested sd disk layout
+cattlepi_check_apply_disk_layout()
+{
+    cattlepi_fetch_update_config
+    jq -r ".config.sdlayout" "${CATTLEPI_DIR}/config" | grep -q null
+    if [ $? -ne 0 ]; then
+        # pull down and decode the requested disk layout
+        jq -r ".config.sdlayout" "${CATTLEPI_DIR}/config" | base64 -d > /tmp/sdlayout
+        # TODO: there should be proper error checking here, but do the bare minimum here
+        # dump current config
+        sfdisk -d /dev/mmcblk0 | grep -v label-id: > /tmp/current_sdlayout
+        cmp -s /tmp/sdlayout /tmp/current_sdlayout
+        if [ $? -ne 0 ]; then
+            # we need to apply the requested layout
+            #   to keep things "bootable" we will backup the content under /boot, apply the layout
+            #   and afterwards copy back the backup to the boot partition
+            mkdir /run/bootbk
+            # TODO: here we will need to remove all the images to prevent copying (and filling the memory as the
+            # /run in on a tmpfs, therefore in memory)
+            cp -R /boot/* /run/bootbk
+            umount /boot
+            # showtime: apply the requested layout
+            sfdisk /dev/mmcblk0 < /tmp/sdlayout
+            # now attempt to mkfs on the newly created partitions
+            sfdisk -J /dev/mmcblk0 > /tmp/sdlayout.json
+            PARTC=$(jq '.partitiontable.partitions | length' /tmp/sdlayout.json)
+            let PARTC=$((PARTC - 1))
+            for IPART in `seq 0 $PARTC`
+            do
+                PARTNODE=$(jq -r ".partitiontable.partitions[$IPART].node" /tmp/sdlayout.json)
+                PARTTYPE=$(jq -r ".partitiontable.partitions[$IPART].type" /tmp/sdlayout.json | )
+                echo "$PARTNODE: type $PARTTYPE"
+                if [ $PARTTYPE == "b" ]; then
+                    mkfs.vfat -F 32 $PARTNODE
```

Comment:
technically this is the partition type. 83 is Linux native partition which can mean ext2,3,4 among other things. b is FAT32
I would say that 83 is by default ext4 and b is FAT32. We can add support for other types and/or add metadata that would allow us to derivce what's the filesystem on a partition but at the end of the day you can pretty much use whatever filesystem you want via mkfs.